### ISSUE TITLE
Log spam:  Unrecognized metric label

### ISSUE THREAD

**Issue by wstrange** [GoogleCloudPlatform/k8s-stackdriver#151]
We are seeing a lot of log entries in stackdriver of the form:

E0509 04:26:19.168197 1 stackdriver.go:58] Error while sending request to Stackdriver googleapi: Error 400: Field timeSeries[0].metric.labels[1] had an invalid value of "code": Unrecognized metric label., badRequest
--

This is on a GKE cluster, 1.9.7-gke.0

This is coming from the prometheus-to-sd container

**Comment by x13n** [comment]
@loburm Can you take a look?

**Comment by vyfster** [comment]
On GKE cluster 1.9.7-gke.0.  I get the same as the above message and the following where `code` is replaced with `grpc` -

```
E0519 09:17:23.051452 1 stackdriver.go:58] Error while sending request to Stackdriver googleapi: Error 400: Field timeSeries[0].metric.labels[0] had an invalid value of "grpc": Unrecognized metric label., badRequest
```

I've updated the yaml to both the fluentd-gcp-v2.0.17 deployment and the yaml to the individual pods changing the prometheus-to-sd-exporter image to `gcr.io/google-containers/prometheus-to-sd:v0.2.6` to no effect. Still get both error messages spamming the logs.

**Comment by tomoe** [comment]
Could this be caused by a prometheus data schema change leading to incompatible TimeSeries resource? 

Here's the diff of the fluentd metrics (v1.8 vs v1.10 GKE clusters):
```
$ diff -u 1.8.txt 1.10.txt 
--- 1.8.txt     2018-05-23 23:54:34.472643397 +0900
+++ 1.10.txt    2018-05-24 00:53:35.567634042 +0900
@@ -1,28 +1,33 @@
 # TYPE process_start_time_seconds gauge
 # HELP process_start_time_seconds Timestamp of the process start in seconds
-process_start_time_seconds 1527087032.0
+process_start_time_seconds 1527086089
 # TYPE logging_entry_count counter
 # HELP logging_entry_count Total number of log entries generated by either application containers or system components
-logging_entry_count 459.0
+logging_entry_count 880
 # TYPE stackdriver_successful_requests_count counter
 # HELP stackdriver_successful_requests_count A number of successful requests to the Stackdriver Logging API
-stackdriver_successful_requests_count{grpc="false"} 32.0
+stackdriver_successful_requests_count{grpc="true",code="0"} 48
 # TYPE stackdriver_failed_requests_count counter
 # HELP stackdriver_failed_requests_count A number of failed requests to the Stackdriver Logging API, broken down by the error code
 # TYPE stackdriver_ingested_entries_count counter
 # HELP stackdriver_ingested_entries_count A number of log entries ingested by Stackdriver Logging
-stackdriver_ingested_entries_count 459.0
+stackdriver_ingested_entries_count{grpc="true",code="0"} 721
 # TYPE stackdriver_dropped_entries_count counter
 # HELP stackdriver_dropped_entries_count A number of log entries dropped by the Stackdriver output plugin
+# TYPE stackdriver_retried_entries_count counter
+# HELP stackdriver_retried_entries_count The number of log entries that failed to be ingested by the Stackdriver output plugin due to a transient error and were retried
 # TYPE fluentd_status_buffer_queue_length gauge
 # HELP fluentd_status_buffer_queue_length Current buffer queue length.
-fluentd_status_buffer_queue_length{plugin_id="object:3fe4940f3454",plugin_category="output",type="google_cloud"} 0.0
-fluentd_status_buffer_queue_length{plugin_id="object:3fe4941a8aac",plugin_category="output",type="google_cloud"} 0.0
+fluentd_status_buffer_queue_length{plugin_id="object:160bf98",plugin_category="output",type="google_cloud"} 0
+fluentd_status_buffer_queue_length{plugin_id="object:16ecf98",plugin_category="output",type="google_cloud"} 0
+fluentd_status_buffer_queue_length{plugin_id="object:1663b44",plugin_category="output",type="google_cloud"} 0
 # TYPE fluentd_status_buffer_total_bytes gauge
 # HELP fluentd_status_buffer_total_bytes Current total size of queued buffers.
-fluentd_status_buffer_total_bytes{plugin_id="object:3fe4940f3454",plugin_category="output",type="google_cloud"} 0.0
-fluentd_status_buffer_total_bytes{plugin_id="object:3fe4941a8aac",plugin_category="output",type="google_cloud"} 1640.0
+fluentd_status_buffer_total_bytes{plugin_id="object:160bf98",plugin_category="output",type="google_cloud"} 3057
+fluentd_status_buffer_total_bytes{plugin_id="object:16ecf98",plugin_category="output",type="google_cloud"} 0
+fluentd_status_buffer_total_bytes{plugin_id="object:1663b44",plugin_category="output",type="google_cloud"} 0
 # TYPE fluentd_status_retry_count gauge
 # HELP fluentd_status_retry_count Current retry counts.
-fluentd_status_retry_count{plugin_id="object:3fe4940f3454",plugin_category="output",type="google_cloud"} 0.0
-fluentd_status_retry_count{plugin_id="object:3fe4941a8aac",plugin_category="output",type="google_cloud"} 0.0
+fluentd_status_retry_count{plugin_id="object:160bf98",plugin_category="output",type="google_cloud"} 0
+fluentd_status_retry_count{plugin_id="object:16ecf98",plugin_category="output",type="google_cloud"} 0
+fluentd_status_retry_count{plugin_id="object:1663b44",plugin_category="output",type="google_cloud"} 0
```

I can see `code`, `grpc` for example in `stackdriver_successful_requests_count`.

**Comment by loburm** [comment]
Hi Tomoe,
Thanks for checking, this a root cause of a problem. I'm going to talk with Stackdriver folks about it.

**Comment by 0x80** [comment]
Hi, 

I'm experiencing this problem as well. I upgraded my cluster and node pool versions via the gcloud web interface, but I don't see any option to downgrade... Can I somehow downgrade without having to recreate the deployment?

**Comment by tomoe** [comment]
Looks like the issue has been resolved in my cluster. 
@wstrange, could you confirm and close unless you still see the issue?

**Comment by wstrange** [comment]
Appears to be fixed in my recent cluster (1.10.2-gke.3)
